From 22c57610aa717c161fb77fa259cbeaf621639a41 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Thu, 1 Jul 2021 11:16:42 +0300
Subject: [PATCH 25/29] pci:s32-gen1: Use hardware defaults for PCIe coherency

No longer touch the PCIe coherency settings; the hardware defaults
produce sane-enough settings.

Upstream-Status: Pending

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-7389
---
 drivers/pci/pcie_s32gen1.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/pci/pcie_s32gen1.c b/drivers/pci/pcie_s32gen1.c
index 5e8cbe3895..5ff4017d48 100644
--- a/drivers/pci/pcie_s32gen1.c
+++ b/drivers/pci/pcie_s32gen1.c
@@ -622,13 +622,10 @@ bool s32_pcie_init(void __iomem *dbi, int id, bool rc_mode,
 	if (!s32_pcie_set_link_width(dbi, id, linkwidth))
 		return false;
 
-	/* Configure transactions as Cacheable, Outer Shareable. */
-	s32_pcie_change_mstr_ace_cache(dbi, 3, 3);
-	s32_pcie_change_mstr_ace_domain(dbi, 2, 2);
-
-	/* Test value for coherency control reg */
-	debug("COHERENCY_CONTROL_3_OFF: 0x%08x\n",
-	      in_le32(PCIE_PORT_LOGIC_COHERENCY_CONTROL_3(dbi)));
+	/* PCIE_COHERENCY_CONTROL_<n> registers provide defaults that configure
+	 * the transactions as Outer Shareable, Write-Back cacheable; we won't
+	 * change those defaults.
+	 */
 
 	BSET32(PCIE_PORT_LOGIC_PORT_FORCE(dbi), PCIE_DO_DESKEW_FOR_SRIS);
 
-- 
2.17.1

